language/move-ir/types/src/spec_language_ast.rs

```diff
@@ -60,6 +60,8 @@ pub enum SpecExp {
     Not(Box<SpecExp>),
     /// Binary operators also suported by Move
     Binop(Box<SpecExp>, BinOp, Box<SpecExp>),
+    /// Update expr (i := 1 inside [])
+    Update(Box<SpecExp>, Box<SpecExp>),
```

Comment:
I'm wondering why we need update in an expression? This is very unusual from language design viewpoint.